### ISSUE TITLE
ruby 2.5.7

### ISSUE THREAD

**Issue by dahogenelst** [locomotivecms/steam#159]
I've noticed that a simple

```
      {% with_scope publish_date.lte: today %}
      {% for post in contents.posts %}
        {% include blog_small with post %}
      {% endfor %}
      {% endwith_scope %}
```

didn't work with ruby 2.5.7. Why exactly i wasn't able to figure out

**Comment by did** [comment]
thanks @dahogenelst!!! I should have considered your issue sooner!  We're having a bug in production because of this. I'm looking into it right now!
I guess it's caused by a wrong regex issue.

**Comment by did** [comment]
ruby 2.6.5 is also wrong.

**Comment by did** [comment]
Alright, so more news about it. We rely on Solid, a gem created a while ago as a layer of top of Liquid. One of the good things on Solid is their string parsing mechanism based on the Ruby parser itself.  In other words, we don't have to write complex regexps to parse the options of a liquid tag.
But for some reasons, the new versions of Ruby (2.5.7, 2.6.5 and probably 2.7.x) break their parser. 

And since we haven't updated our Liquid fork for a while, I decided 2 things:
- get rid of the Solid gem
- update our fork to get the last version of the Shopify Liquid. 

I've started a new branch:  https://github.com/locomotivecms/steam/tree/liquid-4.

**Comment by did** [comment]
quick update. I managed to upgrade all the liquid tags / drops and filters to match the last version of the Liquid gem + make them pass all the specs.
I'm testing this branch against client sites that we're developing. If everything goes well, I'll merge this branch soon. 
One last thing, I've to double check in Engine too.

**Comment by did** [comment]
fixed by #160
